nxpprog.py

```diff
@@ -398,6 +399,80 @@
 prog = None
 
 
+LPC40XX_CHAR = {'Question': b'?',
+                'Synchronized': b'Synchronized\r\n',
+                'OK': b'OK\r\n'}
+
+
+def port_write_and_verify(port, payload, error_message="", debug_message=""):
+    bytes_sent = port.write(bytearray(payload))
+    if bytes_sent != len(payload):
+        log(error_message)
+        return False
+    else:
+        log(debug_message)
+        return True
```

Comment:
Change this from returning `True` or `False` to throwing an exception if something goes wrong. Will can be useful to keep us from having to check the result of this function each time something goes wrong.

Reply:
I had `port_write_and_verify` raise `SerialException` with message